src/components/Signup/Signup.jsx

```diff
@@ -0,0 +1,176 @@
+import "./authentication.css";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+
+export const Signup = () => {
+  const navigate = useNavigate();
+  const [email, setEmail] = useState();
+  const [password, setPassword] = useState();
+  const [passwordValue, setPasswordvalue] = useState();
+  const [confirm, setConfirm] = useState();
+
+  const handleSignup = async (event) => {
+    try {
+      event.preventDefault();
+      const { firstname, lastname, email, pass } = event.target.elements;
+      console.log()
+      const response = await axios.post(`/api/auth/signup`, {
+        firstName: firstname.value,
+        lastName: lastname.value,
+        email: email.value,
+        password: pass.value,
+      });
+      if (response.data.encodedToken) {
+        localStorage.setItem(
+          "token",
+          JSON.stringify(response.data.encodedToken)
+        );
+        navigate("/login");
+      }
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  function checkEmail(e) {
+    let regEmail =
```

Comment:
remove this checking email logic to utils so that you can re-use the function

Reply:
Thank you for suggesting, Since I am using this for signup on;y I kept here.